app/src/lib/release-notes.ts

```diff
@@ -83,10 +83,19 @@ export function getReleaseSummary(
   }
 }
 
-export async function getChangeLog(): Promise<ReadonlyArray<ReleaseMetadata>> {
+export async function getChangeLog(
+  limit?: number
+): Promise<ReadonlyArray<ReleaseMetadata>> {
   const changelog =
     'https://central.github.com/deployments/desktop/desktop/changelog.json'
-  const query = __RELEASE_CHANNEL__ === 'beta' ? '?env=beta' : ''
+  const params = new URLSearchParams()
+  if (__RELEASE_CHANNEL__ === 'beta') {
+    params.append('env', 'beta')
+  }
+  if (limit !== undefined) {
+    params.append('limit', limit.toString())
+  }
+  const query = params.toString() !== '' ? `?${params.toString()}` : ''
 
   const response = await fetch(`${changelog}${query}`)
```

Comment:
I personally prefer to handle URLs as I did in https://github.com/desktop/desktop/blob/bbe45dbcb676aad7505cbf7af8f9ba7144b8d8b1/app/src/ui/lib/update-store.ts#L173-L178

It would look something like this here:

```typescript
  const changelog =
    new URL('https://central.github.com/deployments/desktop/desktop/changelog.json')
  const params = new URLSearchParams()
  if (__RELEASE_CHANNEL__ === 'beta') {
    changelog.searchParams.set('env', 'beta')
  }
  if (limit !== undefined) {
    changelog.searchParams.set('limit', limit.toString())
  }

  const response = await fetch(changelog.toString())
```

This way you don't have to worry about concatenating `?` or any of that.

Reply:
:D See I did a brief google search for something like that when I learned of URLSearchParams.. since i figured if there was that there should be a way to for it to know about the `?` and `&`. I will refactor to that tomorrow.

Reply:
@sergiou87 this is even better, today I 🤓

Reply:
Done! Much cleaner - thank you @sergiou87 and @say25!